tests/algorithm/scripts/expand_test_data.py

```diff
@@ -0,0 +1,243 @@
+import copy
+import random
+import re
+import string
+
+BASE_TEST_FILE = "../data/nbs_test.csv"
+EXPANDED_OUTPUT_FILE = "../data/expanded_nbs_test.csv"
+
+ALGORITHM_RELEVANT_COLUMNS = [
+    "BIRTHDATE",
+    "FIRST",
+    "LAST",
+    "SUFFIX",
+    "GENDER",
+    "ADDRESS",
+    "CITY",
+    "ZIP",
+    "SSN"
+]
+
+# Range of the number of copies to make of each test case, before
+# applying scrambling. Value is chosen randomly within the range.
+# Note that the range of how many cases you end up with for each
+# starting case is this range + 1, since you have the original case
+# plus the duplicates (we want to preserve the original test, too)
+MIN_DUPLICATE_CASES = 2
+MAX_DUPLICATE_CASES = 6
+
+# Controls for the random dropout of field information to simulate
+# missingness issues in incoming data. For each duplicate created
+# for a test case, there is a desired probability to apply dropout
+# to some number of fields in that duplicate. If that duplicate is
+# randomly chosen to experience dropout, then a random number of
+# fields in the range specified are blanked out.
+CHANCE_TO_DROP_FIELDS = 0.5
+MIN_FIELDS_TO_DROP = 1
+MAX_FIELDS_TO_DROP = 2
+
+# Controls for scrambling the value of feature fields in a test case
+# duplicate. For each duplicate generated, there is some probability
+# of applying field scrambling. If a duplicate is flagged for scrambling,
+# then the number of fields to scramble is randomly selected. For each
+# randomly selected field, the number of "quality issues" to be 
+# applied (as measured by edits including addition, deletion, and 
+# transposition) is randomly determined from a given range.
+CHANCE_TO_SCRAMBLE = 0.5
+MIN_FIELDS_TO_SCRAMBLE = 1
+MAX_FIELDS_TO_SCRAMBLE = 3
+MIN_EDIT_DISTANCE = 1
+MAX_EDIT_DISTANCE = 2
+
+
+def identify_missing_fields(test_case: dict[str, str]) -> list:
+    """
+    Identify which fields in a given test case, if any, are missing
+    data that might be relevant to the algorithm.
+    """
+    missing_fields = []
+    for field in ALGORITHM_RELEVANT_COLUMNS:
+        if test_case[field] is None or test_case[field] == "":
+            missing_fields.append(field)
+    return missing_fields
+
+
+def apply_field_scrambling(
```

Comment:
I think this method could be really useful for some concepts I've been thinking about for load testing.  The big difference is we'll be dealing with JSON and not CSV input.  No changes necessary, but if you could think of an easy way to abstract the code so its format (json, csv, tsv, etc) agnostic, that would be really helpful.  This is also something we can iterate on in the future.

Reply:
Yeah I suspect we'll need to iterate on this as we get into load testing, when we see what our data shapes and structures look like more precisely, but definitely agree. At its heart this is really an intelligent randomization function that keeps distributions looking reasonable and controls quality, and that's not predicated on data format. We should definitely be able to abstract that randomization to a more general case once we get into it!